packages/fuchsia_remote_debug_protocol/test/src/dart/dart_vm_test.dart

```diff
@@ -0,0 +1,181 @@
+        return new Future<json_rpc.Peer>(() => null);
+      }
+
+      fuchsiaVmServiceConnectionFunction = mockServiceFunction;
+      expect(await DartVm.connect(Uri.parse('http://this.whatever/ws')),
+          equals(null));
+    });
+
+    test('disconnect closes peer', () async {
+      final MockPeer peer = new MockPeer();
+      Future<json_rpc.Peer> mockServiceFunction(Uri uri) {
+        return new Future<json_rpc.Peer>(() => peer);
+      }
+
+      fuchsiaVmServiceConnectionFunction = mockServiceFunction;
+      final DartVm vm =
+          await DartVm.connect(Uri.parse('http://this.whatever/ws'));
+      expect(vm, isNot(null));
+      await vm.stop();
+      verify(peer.close());
+    });
+  });
+
+  group('DartVm.getAllFlutterViews', () {
+    MockPeer mockPeer;
+
+    setUp(() {
+      mockPeer = new MockPeer();
+    });
+
+    tearDown(() {
+      restoreVmServiceConnectionFunction();
+    });
+
+    test('basic flutter view parsing', () async {
+      final Map<String, dynamic> flutterViewCannedResponses = <String, dynamic>{
+        'views': <Map<String, dynamic>>[
+          <String, dynamic>{
+            'type': 'FlutterView',
+            'id': 'flutterView0',
+          },
+          <String, dynamic>{
+            'type': 'FlutterView',
+            'id': 'flutterView1',
+            'isolate': <String, dynamic>{
+              'type': '@Isolate',
+              'fixedId': 'true',
+              'id': 'isolates/1',
+              'name': 'file://flutterBinary1',
+              'number': '1',
+            },
+          },
+          <String, dynamic>{
+            'type': 'FlutterView',
+            'id': 'flutterView2',
+            'isolate': <String, dynamic>{
+              'type': '@Isolate',
+              'fixedId': 'true',
+              'id': 'isolates/2',
+              'name': 'file://flutterBinary2',
+              'number': '2',
+            },
+          },
+        ],
+      };
+
+      Future<json_rpc.Peer> mockVmConnectionFunction(Uri uri) {
+        when(mockPeer.sendRequest(any, any)).thenReturn(
+            new Future<Map<String, dynamic>>(() => flutterViewCannedResponses));
+        return new Future<json_rpc.Peer>(() => mockPeer);
+      }
+
+      fuchsiaVmServiceConnectionFunction = mockVmConnectionFunction;
+      final DartVm vm =
+          await DartVm.connect(Uri.parse('http://whatever.com/ws'));
+      expect(vm, isNot(null));
+      final List<FlutterView> views = await vm.getAllFlutterViews();
+      expect(views.length, 3);
+      // Check ID's as they cannot be null.
+      expect(views[0].id, 'flutterView0');
+      expect(views[1].id, 'flutterView1');
+      expect(views[2].id, 'flutterView2');
+
+      // Verify names.
+      expect(views[0].name, equals(null));
+      expect(views[1].name, 'file://flutterBinary1');
+      expect(views[2].name, 'file://flutterBinary2');
+    });
+
+    test('invalid flutter view parsing', () async {
```

Comment:
Are they invalid or just happen to not be Flutter views? If they are actually invalid, perhaps we should throw an error (after all this will be used for testing Fuchsia).

Reply:
I don't think those are mutually exclusive given the RPC invocation is intended to get only Flutter views.

Reply:
I've updated the code to throw an `RpcFormatError` when receiving a malformed response.